### ISSUE TITLE
Question about ring_client.c

### ISSUE THREAD

**Issue by sarsanaee** [Orange-OpenSource/oko#9]
Hey,

Looking at the `tests/dpdk/ring_client.c`. It seems that you are resuing the bpf examples available in `examples/bpf` like `dapper` in there. However, I don't see any usage of `ubpf_vm` in `tests/dpdk/ring_client.c`. I was expecting the VM to operate in there, right?

At the moment, there is no JITing or interpretation or anything in that file. Is this correct? If yes, would you please point me to some tests that actually run the vm? I'm wondering if that's something we should do at all or not!

Thanks,
Alireza

**Comment by pchaigno** [comment]
> It seems that you are resuing the bpf examples available in `examples/bpf` like `dapper` in there.

Yes, I reused some of the code from the example BPF programs. `ring_client.c` also reuses the code for the BPF maps.

> However, I don't see any usage of `ubpf_vm` in `tests/dpdk/ring_client.c`. I was expecting the VM to operate in there, right?

No, the goal was to compare programs running on DPDK with their equivalent running on Oko (i.e., ubpf on top of Open vSwitch-DPDK). The programs running on DPDK reuse the code for the maps, but they are compiled from C to x86 directly, without using BPF bytecode.

> At the moment, there is no JITing or interpretation or anything in that file. Is this correct? If yes, would you please point me to some tests that actually run the vm? I'm wondering if that's something we should do at all or not!

The example BPF programs at `examples/bpf/` can be compiled and loaded in Oko as per https://github.com/Orange-OpenSource/oko#how-to-use. Is that what you are looking for?
